lib/fast_jsonapi/serialization_core.rb

```diff
@@ -27,6 +27,30 @@ def ids_hash(ids, record_type)
         id_hash(ids, record_type) # ids variable is just a single id here
       end
 
+      def id_hash_from_record(record, record_types)
+        # memoize the record type within the record_types dictionary, then assigning to record_type:
+        record_type = record_types[record.class] ||= record.class.name.underscore.to_sym
+        { id: record.id.to_s, type: record_type }
+      end
+
+      def ids_hash_from_record_and_relationship(record, relationship)
+        polymorphic = relationship[:polymorphic]
+
+        return ids_hash(
+          record.public_send(relationship[:id_method_name]),
+          relationship[:record_type]
+        ) unless polymorphic
```

Comment:
First of all, thanks so much for this PR. I'd really like to use this feature in a project, but I think the implementation here should change a bit in the case of polymorphic belongs_to.

`type_method_name` should be inferred directly from the `object_method_name` in the case of a polymorphic association since this value should be saved directly on the record. Something along the lines of this might make that work (I have yet to test this):

```ruby
type_method_name = polymorphic ? "#{object_method_name}_type" : relationship[:record_type]
return ids_hash(
  record.public_send(relationship[:id_method_name]),
  record.public_send(type_method_name),
) if relationship[:relationship_type] == :belongs_to || !polymorphic
```

This is in opposition to the implementation here of always referring to the `record.class` and sending it to `id_hash_from_record` in the case of a `belongs_to` association. I believe this would save a database query. It would also save having to define a map when your `belongs_to` relationship is different than the underlying record's class name. Thoughts?

Reply:
I think I'll make a PR for this tomorrow if I can get all tests to run. I'd really like to hear if anyone thinks this approach would cause issues.

Reply:
Ok, I tested this out locally and messed it up a little. I'm going to edit my original code sample above for something that worked for me.

Reply:
```ruby
type_name = polymorphic ? record.public_send("#{relationship[:object_method_name]}_type") : relationship[:record_type]

# could be re-written for readability as:

type_name = if polymorphic
  type_method_name = "#{relationship[:object_method_name]}_type"
  record.public_send(type_method_nam)
else
  relationship[:record_type]
end
```

Reply:
Side note: Instead of constructing the `type_name` here, you could also define it within the `fetch_polymorphic_option` method call.